### ISSUE TITLE
[@typescript-eslint/no-unused-vars] Typeguard variable marked as unused

### ISSUE THREAD

**Issue by ambrusadrianzh** [typescript-eslint/typescript-eslint#1011]
**What code were you trying to parse?**

```ts
declare type UserId = string // read as a refined type
const isUserId = (s: string): s is UserId => true
```

**What did you expect to happen?**
`w` should not be marked as unused.
**What actually happened?**
error  'w' is defined but never used @typescript-eslint/no-unused-vars
**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin-tslint` | `2.3.1` |
| `@typescript-eslint/parser`               | `2.3.1` |
| `TypeScript`                              | `3.6.3` |
| `ESLint`                                  | `6.4.0` |
| `node`                                    | `10.15.3` |
| `npm`                                     | `6.11.3` |
